pipeline/projects/knoedler/data.py

```diff
@@ -708,35 +638,34 @@ def add_prev_prev(data: dict, gpi=None):
 		data['prev_uid'] = None
 	return data
 
-@use('uuid_cache')
-def fan_prev_post_purchase_sale(data: dict, uuid_cache=None):
+
+@use('ulan_type')
+def fan_prev_post_purchase_sale(data: dict, ulan_type=None):
 	# One owner = two acquisitions... transfer to, transfer from
 
-	data['owner_uuid'] = fetch_uuid(data['owner_uid'], uuid_cache)
-	data['owner_type'] = get_actor_type(data['owner_ulan'], uuid_cache)
-	data['object_uuid'] = fetch_uuid(data['object_uid'], uuid_cache)
+	data['owner_uid'] = data['owner_uid']
+	data['owner_type'] = ulan_type.get(data['owner_ulan'], "Person")
+	data['object_uid'] = data['object_uid']
 	if 'prev_uid' in data and data['prev_uid']:
-		data['prev_uuid'] = fetch_uuid(data['prev_uid'], uuid_cache)
+		data['prev_uid'] = data['prev_uid']
 
 	for t in ['purchase', 'sale']:
 		data = data.copy()
 		data['uid'] = "%s_%s" % (data['rowid'], t)
-		data['uuid'] = fetch_uuid(data['uid'], uuid_cache)
 		data['acq_type'] = t
 		if t == 'sale':
 			data['prev_uid'] = "%s_purchase" % data['rowid']
-			data['prev_uuid'] = fetch_uuid("%s_purchase" % data['rowid'], uuid_cache)
 
 		yield data
 
 
 # ~~~~ People Tables ~~~~
 
 @use('gpi')
-@use('uuid_cache')
-def add_person_names(thing: dict, gpi=None, uuid_cache=None):
+def add_person_names(thing: dict, gpi=None):
 	s = 'SELECT * FROM gpi_people_names WHERE person_uid = :id'
 	thing['names'] = []
+	thing['uri'] = f"{UID_TAG_PREFIX}{thing['uid']}"
```

Comment:
Is `thing['uid']` guaranteed to be OK here, or does it need URL escaping?

Reply:
It should be okay -- it's only ever [a-zA-Z0-9-] as far as I know.

Also, it'll never see the light of day, as we'll just subst it out in the rewrite script.

Reply:
FWIW, I've found it *incredibly* helpful for debugging to be able to work with the resulting JSON-LD data *without* rewriting to UUIDs. And while most of our tooling seems perfectly happy with malformed URIs, I'm not sure that means we should be careless about it. (And if you want to push that debugging work into something like Jena, then it definitely will become a problem.)

Reply:
True enough!
Should refactor a bit to have the construction handled in a function rather than just a string formatter to ensure URI compatibility.

Should rewrite then unescape them before substitution, or should it expect to work from the escaped form? Either way, we should document that clearly as an obvious gotcha

Reply:
> True enough!
> Should refactor a bit to have the construction handled in a function rather than just a string formatter to ensure URI compatibility.

Yeah. Here's what I did in PIR: https://github.com/thegetty/pipeline/blob/52d12db9d13cbac9037a3a9b0e6e438b15821608/pipeline/projects/provenance.py#L68

> Should rewrite then unescape them before substitution, or should it expect to work from the escaped form? Either way, we should document that clearly as an obvious gotcha

I'm not sure I understand the question here. The tag URIs should always be valid URIs, so escaped at construction time. I don't think there's ever a reason to unescape them.

Reply:
Okay - the thought was if someone is working from the raw data to pre-construct a UUID cache, then they would also need to know to escape the UIDs. Hence needing a decision and documentation somewhere to that end.

Reply:
Ah, yes. That's true. For anything coming in from outside the pipeline, it will either need to be pre-encoded, or we'll need a processing step after the JSON is read (which wouldn't be terrible as it would solve my 'dict using tuples as keys` issue).